lambdas/revoke-vcs/src/main/java/uk/gov/di/ipv/core/revokevcs/RevokeVcsHandler.java

```diff
@@ -0,0 +1,108 @@
+    private final VerifiableCredentialService verifiableCredentialService;
+    private final DataStore<VcStoreItem> dataStore;
+
+    @SuppressWarnings("unused") // Used by AWS
+    public RevokeVcsHandler(
+            ConfigService configService,
+            VerifiableCredentialService verifiableCredentialService,
+            DataStore<VcStoreItem> dataStore) {
+        this.configService = configService;
+        this.verifiableCredentialService = verifiableCredentialService;
+        this.dataStore = dataStore;
+    }
+
+    @SuppressWarnings("unused") // Used through dependency injection
+    public RevokeVcsHandler() {
+        this.configService = new ConfigService();
+        this.verifiableCredentialService = new VerifiableCredentialService(configService);
+        boolean isRunningLocally = this.configService.isRunningLocally();
+        this.dataStore =
+                new DataStore<>(
+                        this.configService.getEnvironmentVariable(
+                                EnvironmentVariable.REVOKED_USER_CREDENTIALS_TABLE_NAME),
+                        VcStoreItem.class,
+                        DataStore.getClient(isRunningLocally),
+                        isRunningLocally,
+                        configService);
+    }
+
+    @Override
+    @Tracing
+    @Logging(clearState = true)
+    public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context)
+            throws IOException {
+        LogHelper.attachComponentIdToLogs(configService);
+        List<UserIdCriIdPair> userIdCriIdPairs =
+                new ObjectMapper()
+                        .readValue(inputStream, RevokeRequest.class)
+                        .getUserIdCriIdPairs();
+        LOGGER.info("Revoking {} VCs", userIdCriIdPairs.size());
+
+        revoke(userIdCriIdPairs);
+        LOGGER.info("Finished attempt to revoke {} VCs", userIdCriIdPairs.size());
+    }
+
+    private void revoke(List<UserIdCriIdPair> userIdCriIdPairs) {
+        for (int i = 0; i < userIdCriIdPairs.size(); i++) {
+            var userIdCriIdPair = userIdCriIdPairs.get(i);
+            LOGGER.info("Processing VC {} / {}", i, userIdCriIdPairs.size());
+
+            try {
+                revoke(userIdCriIdPair);
+                LOGGER.info("Successfully revoked VC.");
+            } catch (Exception e) {
+                LOGGER.error("Failed to revoke VC.");
+            }
+        }
+    }
+
+    public void revoke(UserIdCriIdPair userIdCriIdPair) {
+        // Read KBV VC for the ID
+        VcStoreItem vc =
+                this.verifiableCredentialService.getVcStoreItem(
+                        userIdCriIdPair.getUserId(), userIdCriIdPair.getCriId());
+
+        if (vc != null) {
+            // Archive VC
+            dataStore.create(vc);
+
+            // Submit audit event
+            // ...
+
+            // Delete VC from the main table
+            // ...
+
+        } else {
+            LOGGER.error("VC cannot be found");
```

Comment:
Note that the logic above will still log a success in this case - that might be fine, but I think we want to make it easy to count all the different cases.

(e.g. it might be preferable to log _either_ 'successfully deleted', 'cannot be found', or 'unexpected error occurred' for each record)

Reply:
The audit event piece of work does more with this because we then send an audit event